ts/screens/services/ServicesHomeScreen.tsx

```diff
@@ -543,29 +579,32 @@ class ServicesHomeScreen extends React.Component<Props, State> {
           title: I18n.t("services.title"),
           body: () => <Markdown>{I18n.t("services.servicesHelp")}</Markdown>
         }}
-        isSearchAvailable={this.props.userMetadata !== undefined}
+        isSearchAvailable={userMetadata !== undefined}
         searchType={"Services"}
       >
-        <NavigationEvents
-          onWillFocus={() => this.setState({ isLongPressEnabled: false })}
-        />
-        {this.props.isSearchEnabled ? (
+        <NavigationEvents onWillFocus={this.onNavigation} />
+        {!this.state.toastContent && pot.isError(potUserMetadata) ? (
```

Comment:
This condition is becoming huge
Could we create a method and use it instead?

```typescript
private renderError = () => {
    const {
      potUserMetadata,
      refreshUserMetadata,
      visibleServicesContentLoadState,
      visibleServicesMetadataLoadState,
      refreshServices
    } = this.props;
    if (!this.state.toastContent && pot.isError(potUserMetadata)) {
      return this.renderErrorPlaceholder(refreshUserMetadata);
    }
    if (
      this.state.toastContent &&
      !pot.isLoading(this.props.potUserMetadata) &&
      (pot.isError(visibleServicesContentLoadState) ||
        pot.isError(visibleServicesMetadataLoadState))
    ) {
      return this.renderErrorPlaceholder(refreshServices);
    }
    return undefined;
  };
```
It should be
```typescript
{errors ? (
          errors
        ) : this.props.isSearchEnabled ? (
          this.renderSearch()
        ) : (
          <React.Fragment>
            <ScreenContentHeader
              title={I18n.t("services.title")}
              icon={require("../../../img/icons/services-icon.png")}
              fixed={Platform.OS === "ios"}
            />
            {this.props.isFisrtServiceLoadCompleted || this.state.toastContent
              ? this.renderTabs()
              : this.renderServiceLoadingPlaceholder()}
            {this.state.isLongPressEnabled &&
              this.renderLongPressFooterButtons()}
          </React.Fragment>
        )}
```

Reply:
updated as suggested within https://github.com/teamdigitale/io-app/pull/1352/commits/1b3102012c6f3b292978180a039d11649fc964d0